java/gateway-core/src/test/java/com/avaulta/gateway/rules/JsonSchemaFilterUtilsTest.java

```diff
@@ -193,21 +189,41 @@ void filterBySchema_refs() {
                 "    \"timestamp\" : \"2023-01-16T05:12:34Z\"\n" +
                 "  }\n" +
                 "}",
-            objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(schemaRuleUtils.filterObjectBySchema(ComplexPojoPlus.builder()
+            objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(jsonSchemaFilterUtils.filterObjectBySchema(ComplexPojoPlus.builder()
                 .simplePojo(simplePlus)
                 .additionalSimplePojo(simplePlus)
                 .build(), schemaWithRefs)));
 
     }
 
+    @SneakyThrows
+    @Test
+    public void compactCopy() {
+        JsonSchemaFilterUtils.JsonSchemaFilter schemaWithRefs =
+            jsonSchemaFilterUtils.generateJsonSchemaFilter(ComplexPojo.class);
+        JsonSchemaFilterUtils.JsonSchemaFilter compactCopy = jsonSchemaFilterUtils.compactCopy(schemaWithRefs);
+
+        assertEquals(ComplexPojo.EXPECTED_COMPACT_SCHEMA,
+            objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(compactCopy));
+
+        // 30% reduction in yaml size
+        assertEquals(404, yamlMapper.writeValueAsString(schemaWithRefs).length());
+        assertEquals(286, yamlMapper.writeValueAsString(compactCopy).length());
+
+        // non-pretty JSON is actually more compact than the yaml (pretty JSON is a lot less compact)
+        // (so if going to give up on human readability by base64gzip'ing the schema, might as well
+        //  do that with the JSON)
+        // eg, https://developer.hashicorp.com/terraform/language/functions/base64gzip
```

Comment:
well, if we go this way, probably there's no point in doing all this compacting code. Just upload the full schema compressed, which make sense.
We might have special lambda calls to output rules, env variables back so can easy check config despite being compressed

Reply:
yeah, i prefer to keep human readable if we can, so people can just see the value in their SSM parameters to have confidence in it.  It's possible base64gzip would be conditional just for the extreme cases (eg, if > 8KB compacted it then gzip)